content/updates/terraform-controller-overview.md

```diff
@@ -0,0 +1,210 @@
+
+A few initial steps need to be taken before we actually can talk about GitOpsifying our infrastructure. First, set up the infrastructure on AWS: creating AWS account(s), setting up a VPC, an EKS cluster, and required IAM resources. Next, as Flux plays a pivotal role in this setup, we install it to the cluster [with Terraform resource `flux_bootstrap_git` using the Flux provider](https://fluxcd.io/flux/installation/#bootstrap-with-terraform). The bootstrap procedure commits Flux components (including a Kustomization manifest) to a desired Git repository and configures synchronization to the same.
+
+![Terraform Controller components in the Git repository](/img/updates/terraform-controller-overview/tf_controller_1.png)
+
+At this point, we're ready to push the Controller manifests from the [release location](https://raw.githubusercontent.com/weaveworks/tf-controller/main/docs/release.yaml) to the repository and watch how it's deployed to the cluster to the namespace `flux-system`.
+
+![Terraform Controller and Flux pods running in the cluster](/img/updates/terraform-controller-overview/tf_controller_2.png)
+
+
+We won't emphasize organizing the repository here since tailoring the manifests' structure should meet your organization's specific needs. However, it's worth mentioning that we aimed to maintain clarity and readability in our code, striving to adhere to the DRY principle as much as possible.
+
+Possible sources for your inspiration are:
+
+* [Ways of structuring your repositories](https://fluxcd.io/flux/guides/repository-structure/) by Flux team.
+* [TF-controller demo: GitOps EKS node group](https://github.com/tf-controller/eks-scaling) by the Controller developers.
+
+Both examples rely on Kustomize, which we've found somewhat inflexible regarding manifest templating. However, this method could certainly be highly effective for other organizations.
+
+## Managing Infrastructure-as-Code with Terraform Controller
+
+At its core, the Terraform Controller consists of two main elements: the Controller itself and the Runners.
+
+The **Controller** is the brain of the operation. It monitors Kubernetes for specific resource definitions and acts upon changes. When a new Terraform resource is defined in Kubernetes, the Controller picks it up and initiates the necessary actions.
+
+**Runners**, on the other hand, are the workhorses. Each Runner is a "disposable" pod (see `*-tf-runner` pod on the screenshot below) that executes the Terraform plans generated by the Controller.
+
+![Runner of our ECR managed by Terraform Controller](/img/updates/terraform-controller-overview/tf_controller_3.png)
+
+
+The example of the Runner logs:
```

Comment:
@gberenice this image is pretty small type in the website preview. Maybe we just include a few lines of the logs in a code block? What do you think?

Reply:
Updated. Let me know if that looks better to you.